Java/core/src/main/java/com/amazon/randomcutforest/returntypes/ConditionalSampleSummary.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.returntypes;
+
+import static com.amazon.randomcutforest.CommonUtils.checkArgument;
+import static com.amazon.randomcutforest.CommonUtils.toFloatArray;
+
+public class ConditionalSampleSummary {
+
+    /**
+     * a collection of typical points (in the shingled, input space) seen from the
```

Comment:
"typical" is not very descriptive. Given that this is a container class, Can we just call this class `Sample`, and we can call this field `SampledPoints`. The fact that the imputation method fills this container with "typical" points is a property of the imputation algorithm, not of the container.

Reply:
Changed the comment -- it is supposed to be "typical" in the information theory sense, basically ion the samples correspond to union of two well-separated gaussians, we should see the two centers. That corresponds to the summarization aspect.